funcframework/framework.go

```diff
@@ -212,8 +212,16 @@ func writeHTTPErrorResponse(w http.ResponseWriter, statusCode int, status, msg s
 	if !strings.HasSuffix(msg, "\n") {
 		msg += "\n"
 	}
+	fmt.Fprintf(os.Stderr, msg)
+
+	// Flush stdout and stderr when running on GCF. This must be done before writing
+	// the HTTP response in order for all logs to appear in Stackdriver.
+	if os.Getenv("K_SERVICE") != "" {
```

Comment:
Should this just be the default?

Reply:
That's true, `K_SERVICE` is a Knative thing right? I did it this way because it's a peculiarity of how GCF handles logs differently for normal and `crash` responses.

Reply:
Agreed. This way there won't be superfluous lines.

Reply:
Anyone have strong opinions?

Reply:
I don't have opinions on this if case when K_SERVICE is not nil.

I've re-opened a related issue that Julie mentioned for the superfluous empty lines – hopefully we can still flush the logs but have a better local devX: https://github.com/GoogleCloudPlatform/functions-framework-go/issues/8

Reply:
No, I was concerned about using `K_SERVICE` to detect that the function running on GCF since it's not a GCF specific env var. The main use case for the FF (for now) is GCF so I think this is fine.

Reply:
Grant: Sounds good. Since the commit on that issue uses this env var too, let's move ahead with this change?
Huy: Makes sense. It would be nice if there were a more robust way to detect environment.